### PR TITLE
Refactor Ansible remediations that search local file systems

### LINKED ISSUE #10893: [affects stabilization] Ansible Playbooks contain not permitted shell commands
#### Description of problem:
During productization run of the downstream test `/CoreOS/scap-security-guide/Sanity/ansible-allowed-shell-commands` we have found that a generated Ansible Playbook contains shell commands that aren't allowed.

#### SCAP Security Guide Version:
Current upstream master branch as of 2023-07-21 as of HEAD a96ccb9

#### Operating System Version:
RHEL 9, RHEL 8 , RHEL 7

#### Steps to Reproduce:

1.` oscap xccdf generate fix --fix-type ansible --output <output.yml> <ds.xml>`

#### Actual Results:

Playbooks contains this inside:

```
grep -e '^\s*SHA_CRYPT_MIN_ROUNDS\s\+' /etc/login.defs || true
df --local | awk '{print $6}' | grep -v Mounted | grep -v '^/dev' || true
```

#### Expected Results:
No  failing output is produced by the downstream test .

#### Additional Information/Debugging Steps:
There was a similar issue in past (https://github.com/ComplianceAsCode/content/issues/10852) but this productization was run after this issue had been fixed.

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

There are some rules which need to locate files and directories in a system but should only consider local files and directories, ignoring remote file systems and some known local directories with pseudo file systems.
Recently the Ansible approach in `dir_perms_world_writable_root_owned` and `no_host_based_files` was refactored in order to make the remediation more robust and more efficient.
A third rule (`no_user_host_based_files`) was discovered with the same issue.
Therefore a macro was created to make the remediation aligned among all relevant rules and to make the same approach easier to be adopted by future rules.

This PR also refactors the Ansible remediation in `set_password_hashing_min_rounds_logindefs` rule by removing unnecessary tasks and align the playbook to the project Style Guides. Finally, the test scenarios were also improved.

#### Rationale:

- More robust, efficient and scalable Ansible remediation.
- Fixes #10893

#### Review Hints:

Automatus tests using Ansible remediation should be enough.

### CODE REVIEW

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_root_owned/ansible/shared.yml [hunk 1]
```diff
@@ -4,64 +4,11 @@
...
-  loop: "{{ result_relevant_root_dirs.files }}"
-
-- name: "{{{ rule_title }}} - Increment Search Paths List with Local Partitions Mount Points"
-  ansible.builtin.set_fact:
-    search_paths: "{{ search_paths | union([item.mount]) }}"
-  loop: '{{ ansible_mounts }}'
-  when:
-    - item.fstype not in excluded_fstypes
-    - item.mount != '/'
+{{{ ansible_create_list_of_local_paths(list_name="search_paths") }}}
```
Comment (jan-cerny) [inline linux_os/guide/system/permissions/files/dir_perms_world_writable_root_owned/ansible/shared.yml:7]: I think that the fail of the test scenario `world_writable_dir_on_nonlocal_fs.fail.sh` on the CI is caused by the invocation of `systemctl` command in this scenario, which hits the problem that there is no Dbus in the container back end environment that we use in CI.

```
+ systemctl restart nfs-server
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```

When executed local with a virtual machine back end the test is passing.

```
[jcerny@fedora scap-security-guide{pr/10912}]$ python3 tests/automatus.py rule --remediate-using ansible --libvirt qemu:///system ssgts_rhel9 dir_perms_world_writable_root_owned
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-07-27-1438/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_dir_perms_world_writable_root_owned
INFO - Script world_writable_dir_on_nonlocal_fs.fail.sh using profile (all) OK
INFO - Script world_writable_dir_owned_by_uid_2.fail.sh using profile (all) OK
INFO - Script all_dirs_ok.pass.sh using profile (all) OK
```

I think the problem isn't related to the contents of this PR.

File: linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_min_rounds_logindefs/ansible/shared.yml [hunk 2]
```diff
@@ -4,28 +4,16 @@
...
     replace: '\g<1>5000\g<2>'
     backup: no
 
-- name: Check to see if SHA_CRYPT_MIN_ROUNDS is explicitly configured
-  shell: |
-    set -o pipefail
-    grep -e '^\s*SHA_CRYPT_MIN_ROUNDS\s\+' /etc/login.defs || true
-  register: check_sha_crypt_min_rounds_result
-
-# NOTE(gyee): there's a possibility that the value of SHA_CRYPT_MIN_ROUNDS is
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_min_rounds_logindefs/ansible/shared.yml:20]: Was the comment useful?

File: linux_os/guide/services/obsolete/r_services/no_user_host_based_files/ansible/shared.yml [hunk 3]
```diff
@@ -3,26 +3,28 @@
...
-  with_items: "{{ shosts_locations.results | map(attribute='files') | list }}"
-  when: shosts_locations is success
+- name: "{{{ rule_title }}} - Define Rule Specific Facts"
+  ansible.builtin.set_fact:
+    user_shosts_files:
+      - /.shosts
+
+- name: "{{{ rule_title }}} - Find All .shosts Files in Local File Systems"
+  ansible.builtin.command:
+    cmd: find {{ item }} -xdev -type f -name ".shosts"
```
Comment (jan-cerny) [inline linux_os/guide/services/obsolete/r_services/no_user_host_based_files/ansible/shared.yml:16]: Will this be reported as not permitted shell command?

Comment (marcusburghardt) [reply linux_os/guide/services/obsolete/r_services/no_user_host_based_files/ansible/shared.yml:16]: No. It is using the `cmd` module and the test only cares about `shell` module, so it won't be reported as an issue.

File: linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_min_rounds_logindefs/ansible/shared.yml (hunk 2 above)
Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_min_rounds_logindefs/ansible/shared.yml:20]: `SHA_CRYPT_MIN_ROUNDS` is specific for SHA-512-based passwords so it is not supported in all cases. In most of the cases (and distros) it is not expected to be set. I think the comments were unnecessary and also the case is now represented in test scenarios. IMO the comment won't be missed.

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**